### ISSUE TITLE
Torrentbytes doesn't work at all

### ISSUE THREAD

**Issue by joq3** [cardigann/cardigann#220]
### Issue experienced
When I add Torrentbytes.net to Cardigann and input my username and password, and press Test.
This is what happens:
An error occurred
No input found with name 'username'.
And sometimes it says password instead of username.

Tried adding ThePirateBay instead, and it works great, so problem lies with Torrentbytes.net (which works with Jackett though).

### Steps to reproduce
Add Torrentbytes.net and press test.

### Cardigann version (in the footer, or via `cardigann --version`)
1.9.3

**Comment by mmarquezs** [comment]
Torrentbytes website has issues lately for example right now I can't even access it it keeps loading in the browser so I'm 99% sure that's the problem. 
(while writing this it loaded but after a loooong time)

**Comment by joq3** [comment]
That's true, noticed they had some issues this morning, but right now I can access it instantly. Yes, true, but I did enter the correct password and I am able to login from the same IP via webbrowser.

I will wait for a while and see if it starts working, I'll return with the results!

Thank you!

**Comment by joq3** [comment]
The site is now working, and I managed to get it working with Jackett, so there has to be a problem with this application?

Any ideas?

**Comment by mmarquezs** [comment]
You're still getting that error or an other one. I have been testing the definition, logs in with the user just fine. The website of Cardigann reports error  when testing (Form not found matching expr 'form'.) testing from the commandline passes the test too and loggin actually works I have checked the cached ages and it's logged in. The issue is not login at least for me but when searching it returns nothing I have to check what's wrong.

**Comment by mmarquezs** [comment]
This fixes the search not returning any result: https://github.com/cardigann/cardigann/pull/223
The website keeps saying: Form not found matching expr 'form'. not sure what causes that the command line test works and the search too.

**Comment by joq3** [comment]
Yes, exactly, I still get this error when pressing Test:

> An error occurred
No input found with name 'username'.

I have tried deleting the tracker and adding it again, confirming the username and password several times (i am not banned). And tested with The Piratebay, which works (so no problem with Cardigann connecting to trackers).

How do I update to that fix you just posted (sorry for a newbie question) :)

**Comment by mmarquezs** [comment]
Well you can wait the fix to be added to the repo and the do:
` cardigann update --channel=edge `
To have the latest version or:
` cardigann update --channel=stable `
for the stable one but I don't know when the stables are released so that might take a while until the fix is there.

If you want to test/use it right now you can download this file: 
https://cdn.rawgit.com/mmarquezs/cardigann/9b7ad5d88bbb35ff6e2c4ea1c9f665a0b1fe937a/definitions/torrentbytes.yml

Put it in your definitions folder, it my case I run it with root user (probably not recommended :P ) and is /root/.config/cardigann/definitions then you can restart the service with `cardigan service restart` or using whatever your system uses to manage the services adn it should use this new version. But rememeber that if in the future a new version of the definition appears this old file will be used instead of the new version you will have to remove it to use newer versions. 

After the restart if you click over the tracker name it should show that it's using the file like this:
![](http://i.imgur.com/6gq7pBe.png)

Now you can try to search to check if it works, don't do the test try directly to search. 
Also you can run the test on the terminal on the same folder of the file:
cardigann test torrentbytes.yml

**Comment by joq3** [comment]
I put the file in /root/.config/cardigann/definitions (although I had to create the folder definitions as there wasn't any). Then did a restart of the service. But clicking TorrentBytes shows me this:
builtin:/definitions/torrentbytes.yml
Which means it doesn't load the new one.

What did I do wrong? Is there another place for definitions?

**Comment by mmarquezs** [comment]
> EDIT: There was another folder here: /.config/cardigann/definitions

That also happened to me I had to change something so it uses the other folder but that's the folder is using now in you computer so for now you can use it it will be fine. (Issue: https://github.com/cardigann/cardigann/issues/47 )

> First time I tried searching it worked, next time it didn't and I get it to work 1 time after restart.

That's weird... I have been trying multiple times without issues :S 

> If I do the command line test, this is what I get: [...]

It's complaining that it doesn't know the username to use for login. This might be either because when you did the command you had removed the tracker from cardigann so it doesn't have the user/password stored or the commandline it's using an other config folder which probably is using the .config folder of the user who executes it.

**Comment by lxt2** [comment]
Merged the PR, can this be closed now?

**Comment by ldexterldesign** [comment]
@lxt2 hey, no, I think this is still problematic (i.e. I can't get it added) - let me know if you want me to create a new bug report and reference related (open) issues?

Kind regards

**Comment by jlongton** [comment]
Just started getting the same error message with Alpharatio.  Haven't changed password or anything.  Still able to log into Alpharatio.  Tried disabling and re-enabling.  Tried starting and stopping app, which is running on QNAP TS-251+.

**Comment by ldexterldesign** [comment]
BUMP

**Comment by joq3** [comment]
@lxt2 just updated to 1.9.4-39-g0d73bd3 to try it out. Still doesn't work. But if I put the yml @mmarquezs posted for Torrentbytes it works with this version. Please use this yml as standard.

Thank you!

**Comment by joq3** [comment]
It still acts a bit weird. It stops working after a while. Cardigann is running and Sonarr too. But it stops finding things, and I don't know how to fix it, sometimes it starts working after a service reboot, or a reboot of the server, or an update. I always check to see if Torrentbytes is online and working, and it is.

Any ideas?

**Comment by jlongton** [comment]
I gave up and switched to Jackett.

**Comment by joq3** [comment]
@jlongton Yes, I used Jackett before, but it did crash all the time for me, when it was running it worked, but I had a crash every 3 days.

**Comment by jlongton** [comment]
I can't remember what prompted me to switch to Cardigann.  It may have been a similar issue.  Cardigann was working much better until recently.  I use both now...each seems to behave ok with certain trackers.

**Comment by ldexterldesign** [comment]
BUMP

**Comment by ldexterldesign** [comment]
BUMP
